### PR TITLE
[ANCHOR-394] Add whitelist to SEP-10 and add the client name field.

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

- Add `client_allow_list` to SEP-10
- Add name field to client config
- When `client_allow_list` is not defined, the domains of all NONCUSTODIAL clients are allowed if `domain` is defined.

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java [hunk 1]
```diff
@@ -155,14 +164,14 @@ void validateConfig(Errors errors) {
 
   void validateClientAttribution(Errors errors) {
     if (clientAttributionRequired) {
-      if (ListHelper.isEmpty(getClientAttributionAllowList())) {
+      if (ListHelper.isEmpty(getDefaultAllowClientDomain())) {
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java:167]: What if anchor support only custodial wallets?

**Review by Ifropc — APPROVED** [review]

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt [hunk 2]
```diff
@@ -75,19 +78,44 @@ class Sep10ConfigTest {
...
     config.validateClientAttribution(errors)
     assertFalse(errors.hasErrors())
-    println(config.clientAttributionAllowList)
-    println(config.knownCustodialAccountList)
   }
 
   @Test
-  fun `test both client attribution deny and allow lists are empty`() {
-    // Create a cofnig with empty ClientsConfig lists
+  fun `test validation of empty client allow list when client attribution is required`() {
```
Comment (philipliu) [inline platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt:84]: Can we add tests for all the validations?

Comment (lijamie98) [reply platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt:84]: Call `config.validate()` only tests the all validations are called in `config.validate()` function. If we think it is necessary, a test that `config.validate()` call all validation can be added. 

This is unit testing the `validateClientAttribution` function.

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java (hunk 1 above)
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/config/PropertySep10Config.java:167]: The validation is only called when the client attribution is used. The custodial wallet does not require client attribution.

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt (hunk 2 above)
Comment (lijamie98) [reply platform/src/test/kotlin/org/stellar/anchor/platform/config/Sep10ConfigTest.kt:84]: If we still feel strong about this, we can add another test that `config.validate()` calls all validations in the future PR.

**Pull request merged**